syntax = "proto3";

package recoverer;

service recover_service{
    rpc TellVersion(Version) returns (Reply);
    rpc Chunk2Send(Image) returns (ChunkList);
    rpc SendChunk(Chunk) returns (Reply);
    rpc KeepAlive(Reply) returns (Reply);
    rpc RecoverServ(ImageAndServName) returns (Reply);
}

message Version {
    int32 image = 1;
    int32 version = 2;
    int32 size = 3;
}

message Reply {
    int32 status = 1;
}

message Image {
    int32 image = 1;
}

message ImageAndServName {
    int32 image = 1;
    string servname = 2;
}

message Chunk {
    int32 image = 1;
    int32 version = 2;
    int32 number = 3;
    bytes data = 4;
    int32 checksum = 5;
}

message ChunkList{
    repeated int32 needed = 1;
}